### PR TITLE
Veli | Week2

### PR DESCRIPTION
**Pull request by veliataseven**
I have just two file for week2 which is ;
index.html and App.js => https://github.com/HackYourFuture/React/commit/685b43ae6c3df709214299f5e62bc6846e6eb262

### CODE REVIEW

File: homework-submission/homework-week1/homework-week2/App.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+const Button = ({ type, text, className }) => {
+  return (
+    <button type={type} className={className}>
+      {text}
+    </button>
+  );
+};
+
+const DynamicList = ({ List, clickHandler, className }) => {
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: As list is not a component it should be with a small letter l(list).

File: homework-submission/homework-week1/index.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    .itemCompleted {
+        text-decoration: line-through;
+      }
+      .itemNotCompleted {
+        text-decoration: none;
+      }
+    </style></style>
+  </head>
+  <body>
+    <div id="thisIsWhereReactIsInjected"></div>
```
Comment (Michielab) [inline homework-submission/homework-week1/index.html]: When fallowing naming conventions the id is better called root.

File: homework-submission/homework-week1/homework-week2/index.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    flex-direction: column;
+    flex-wrap: wrap;
+    width: 30%;
+    padding: 10px;
+    text-align: center;
+    }
+  </style></style>
+  </head>
+  <body>
+    <div id="thisIsWhereReactIsInjected"></div>
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/index.html]: When following naming conventions the id is better called root.

File: homework-submission/homework-week1/homework-week2/index.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    margin: auto;
+    background-color: rgb(245, 245, 245);
+    box-sizing: border-box;
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    width: 30%;
+    padding: 10px;
+    text-align: center;
+    }
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/index.html]: maybe you can use a separate file for the styles and import it here in the index?

File: homework-submission/homework-week1/homework-week2/App.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    });
+    this.setState({ todoList: newList });
+  };
+
+  render() {
+    return (
+      <div className="wrapper">
+        <AddingListItem submitHandler={this.addItem} />
+        <DynamicList
+          List={this.state.todoList}
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: List should be with a small l as it's not a component. I also like to destructure everything from state in the render function and then just use the declared var. Look at previous comment.

File: homework-submission/homework-week1/homework-week2/App.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  removeItem = event => {
+    const idToRemove = event.target.id;
+    const newList = this.state.todoList.filter(item => item.id != idToRemove);
+    newList.forEach((item, index) => {
+      item.id = index;
+    });
+    this.setState({ todoList: newList });
+  };
+
+  render() {
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: const { todoList } = this.state

File: homework-submission/homework-week1/homework-week2/App.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+      },
+      {
+        id: 2,
+        description: 'Eat breakfast',
+        deadline: '2017-09-09',
+      },
+    ],
+  };
+
+  addItem = event => {
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: There are a lot of things happening in the function. Ideally a function should do one thing. So could be an idea to split it up in smaller functions which you call in this one.

File: homework-submission/homework-week1/homework-week2/App.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+    today = yyyy + '-' + mm + '-' + dd;
+
+    const { descriptionValidation, deadlineValidation } = this.validator(description, deadline);
+    if (!deadlineValidation) deadline = today;
+    if (!descriptionValidation) description = 'Do something';
+
+    if (list.length !== 0) {
+      idItem = list[list.length - 1].id + 1;
+    }
+
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: You can also use a map function: 
```
  const updatedToDoList = list.map(() => {
    return { id: idItem, description: description, deadline: deadline };
  });

```

File: homework-submission/homework-week1/homework-week2/App.js [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+      todoList: copyList,
+    });
+  };
+
+  validator = (description, deadline) => {
+    const regex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+    let deadlineValidation = true;
+    let descriptionValidation = true;
+
+    if (deadline == '' || deadline == null || deadline == undefined || !regex.test(deadline))
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: Is there a particular reason why us use: ```==``` over ```===```? It's safer and also a more of a convention to use ```===```. See also: https://stackoverflow.com/questions/523643/difference-between-and-in-javascript.

File: homework-submission/homework-week1/homework-week2/App.js [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+      deadlineValidation = false;
+
+    if (description == '' || description == null || description == undefined)
+      descriptionValidation = false;
+
+    return { descriptionValidation: descriptionValidation, deadlineValidation: deadlineValidation };
```
Comment (Michielab) [inline homework-submission/homework-week1/homework-week2/App.js]: As the property and the key are the same you can also return like this: 

```
return {descriptionValidation, deadlineValidation}
```

File: homework-submission/homework-week1/homework-week2/App.js (hunk 8 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: I am pushing an new todo item to the end of copyList. I guess dont need using map.

File: homework-submission/homework-week1/homework-week2/App.js (hunk 9 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: Yeah I've done.

File: homework-submission/homework-week1/homework-week2/App.js (hunk 10 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: Yeah. I've done

**Comment by veliataseven** [comment]
Thank You so much @Michielab for feedback. Actually as I finished homework as I have pushed it without checking the some mistakes..

File: homework-submission/homework-week1/homework-week2/App.js (hunk 6 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: okay.

File: homework-submission/homework-week1/homework-week2/App.js (hunk 5 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: okay I have done it

File: homework-submission/homework-week1/homework-week2/App.js (hunk 1 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/App.js]: I have done.

File: homework-submission/homework-week1/index.html (hunk 2 above)
Comment (veliataseven) [reply homework-submission/homework-week1/index.html]: okay

File: homework-submission/homework-week1/homework-week2/index.html (hunk 3 above)
Comment (veliataseven) [reply homework-submission/homework-week1/homework-week2/index.html]: Yeah I have done it

**Comment by ArcoMul** [comment]
Hi @veliataseven, great improvements! The only thing which I am missing, and is quite important is @Michielab's comment on the `addItem` function. There is a lot happening there which makes the code a bit unclear. Things I see happening which _could_ be moved to their own function:
- Getting the description and date out of the form input value
- Generating a date string of the current date

For example, for 'Getting the description and date out of the form input value', this would be a nice way in my opinion:

```js
let { description, deadline } = splitToDescriptionAndDeadline(event.target.listItem.value);
```

Give it a try! :-)

**Comment by veliataseven** [comment]
Okay @ArcoMul Thanks..

**Comment by fedefu** [comment]
Thanks for the feedback @ArcoMul and @Michielab! Can you please do the last implementations @veliataseven so we can close this PR? Thanks!

**Comment by ArcoMul** [comment]
👏

**Pull request closed without merging**